pandas/_libs/lib.pyx

```diff
@@ -642,6 +643,34 @@ def array_equivalent_object(ndarray left, ndarray right) -> bool:
     return True
 
 
+ctypedef fused int6432_t:
+    int64_t
+    int32_t
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def array_equal_fast(
+    ndarray[int6432_t, ndim=1] left, ndarray[int6432_t, ndim=1] right,
```

Comment:
I check locally a little bit, and using a memoryview (`int[:]`) instead of ndarray doesn't seem to give a significant speed up here (I seemed to remember that those could sometimes be faster)